### ISSUE TITLE
Incorrect imports in ESM bundle

### ISSUE THREAD

**Issue by kraenhansen** [mongodb/js-bson#371]
I believe this package's esm bundles are suffering from a bug originating in the [rollup-plugin-commonjs](https://www.npmjs.com/package/rollup-plugin-commonjs) dependency (now [@rollup/plugin-commonjs](https://www.npmjs.com/package/@rollup/plugin-commonjs) - I verified that the bug is still present if upgrading to the latest version of this new package).

I believe the root cause is that describe in the ["External imports in the CommonJS plugin" sub-section of this issue on the Rollup.js repository](https://github.com/rollup/plugins/issues/481):
> **🐞 Bug 5:** For external imports, this plugin will always require the default import.

Because of this, when "long" and "buffer" is imported via `require("long")` and `require("buffer")` in the various sourcefiles and [gets translated to `import long from "long"` and `import buffer from "buffer"`](http://unpkg.com/bson@4.0.4/dist/bson.esm.js) by the commonjs plugin, it produces an invalid bundle, since "buffer" does provide a `default` export.

---

This results in a downstream issue when trying to bundle this package into my own IIFE bundle, using Rollup.
Here is some code to reproduce the issue that I'm seeing:

### package.json
```
{
  "name": "bson-rollup-bug",
  "version": "0.1.0",
  "private": true,
  "main": "main.js",
  "dependencies": {
    "@rollup/plugin-node-resolve": "^8.4.0",
    "bson": "^4.0.4",
    "rollup": "^2.21.0",
    "rollup-plugin-node-builtins": "^2.1.2"
  },
  "scripts": {
    "build": "rollup -c"
  },
  "author": "Kræn Hansen <kraen.hansen@mongodb.com>",
  "license": "ISC"
}
```

### main.js
```
import { ObjectID } from "bson";

export function use() {
    return new ObjectID();
}
```

### rollup.config.js
```
import resolve from "@rollup/plugin-node-resolve";
import builtins from "rollup-plugin-node-builtins";

export default [
    {
        input: "main.js",
        output: [
            {
                file: "bundle.js",
                name: "myBundle",
                format: "iife",
            },
        ],
        plugins: [
            builtins(),
            resolve({
                // browser: true,
                // preferBuiltins: true,
            }),
        ],
    },
];
```

### Output from running `npx rollup -c`

```
> rollup -c

main.js → bundle.js...
[!] Error: 'default' is not exported by node_modules/long/src/long.js, imported by node_modules/bson/dist/bson.esm.js
https://rollupjs.org/guide/en/#error-name-is-not-exported-by-module
node_modules/bson/dist/bson.esm.js (1:7)
1: import long from 'long';
          ^
2: import buffer from 'buffer';
Error: 'default' is not exported by node_modules/long/src/long.js, imported by node_modules/bson/dist/bson.esm.js
    at error (/Users/kraenhansen/Projects/bson-bug/node_modules/rollup/dist/shared/rollup.js:5171:30)
    at Module.error (/Users/kraenhansen/Projects/bson-bug/node_modules/rollup/dist/shared/rollup.js:9603:16)
    at handleMissingExport (/Users/kraenhansen/Projects/bson-bug/node_modules/rollup/dist/shared/rollup.js:9525:28)
    at Module.traceVariable (/Users/kraenhansen/Projects/bson-bug/node_modules/rollup/dist/shared/rollup.js:9986:24)
    at ModuleScope.findVariable (/Users/kraenhansen/Projects/bson-bug/node_modules/rollup/dist/shared/rollup.js:8549:39)
    at MemberExpression.bind (/Users/kraenhansen/Projects/bson-bug/node_modules/rollup/dist/shared/rollup.js:6324:49)
    at AssignmentExpression.bind (/Users/kraenhansen/Projects/bson-bug/node_modules/rollup/dist/shared/rollup.js:2880:23)
    at ExpressionStatement$1.bind (/Users/kraenhansen/Projects/bson-bug/node_modules/rollup/dist/shared/rollup.js:2880:23)
    at Program$1.bind (/Users/kraenhansen/Projects/bson-bug/node_modules/rollup/dist/shared/rollup.js:2876:31)
    at Module.bindReferences (/Users/kraenhansen/Projects/bson-bug/node_modules/rollup/dist/shared/rollup.js:9599:18)
```

**Comment by Jack-Works** [comment]
Also having this problem.

> Uncaught (in promise) TypeError: Failed to resolve module specifier "long". Relative references must start with either "/", "./", or "../".

**Comment by nbbeeken** [comment]
Just for reference we use [JIRA](https://jira.mongodb.org/projects/NODE) to track our issues and prefer that issues are submitted there so we can better prioritize them.
Thanks for submitting! resolved in PR #404
